app/services/spree_shopify_importer/data_parsers/zones/base_data.rb

```diff
@@ -0,0 +1,52 @@
+module SpreeShopifyImporter
+  module DataParsers
+    module Zones
+      class BaseData
+        def initialize(shopify_shipping_zone)
+          @shopify_shipping_zone = shopify_shipping_zone
+        end
+
+        def attributes
+          @attributes ||= {
+            name: @shopify_shipping_zone.name,
+            description: description,
+            kind: @kind
+          }
+        end
+
+        private
+
+        def description
+          "shopify shipping to " + define_members.join(',')
+        end
+
+        private
+        # shopify has not this distinction, there can be zones with coutries and provinces as well
+        def define_spree_zone_kind
+          @kind = count_provinces > 0 ? 'state' : 'country'
+        end
+
+        def count_provinces
+          @shopify_shipping_zone.countries.map do |country|
+            country.attributes['provinces'].count
```

Comment:
may be you can access `provinces` just by `country.provinces`?

As I see in method below, you can.